koku/sources/kafka_listener.py

```diff
@@ -153,10 +153,14 @@ def process_synchronize_sources_msg(msg_tuple, process_queue):
     except Exception as error:
```

Comment:
This generic exception keeps the listener running when some unknown failure happens anywhere in the message processing flow:
```
        # The reason for catching all exceptions is to ensure that the event
        # loop remains active in the event that provider synchronization fails unexpectedly.
```

If you want to use the `SourcesHTTPClient` to set a status, then you need to handle when it raises an exception. The `set_source_status` function itself actually has code to raise an exception. If it does, it breaks the sources-listener.